drivers/gpu/drm/radeon/evergreen_dma.c

```diff
@@ -136,12 +179,19 @@ struct radeon_fence *evergreen_copy_dma(struct radeon_device *rdev,
 		cur_size_in_dw = size_in_dw;
 		if (cur_size_in_dw > 0xFFFFF)
 			cur_size_in_dw = 0xFFFFF;
+		printk("copy dma loop %d of %d: copying 0x%X bytes from 0x%llX to 0x%llX",i,num_loops,cur_size_in_dw*4,src_offset,dst_offset);
 		size_in_dw -= cur_size_in_dw;
 		radeon_ring_write(ring, DMA_PACKET(DMA_PACKET_COPY, 0, cur_size_in_dw));
 		radeon_ring_write(ring, dst_offset & 0xfffffffc);
 		radeon_ring_write(ring, src_offset & 0xfffffffc);
 		radeon_ring_write(ring, upper_32_bits(dst_offset) & 0xff);
 		radeon_ring_write(ring, upper_32_bits(src_offset) & 0xff);
+		/*radeon_ring_write(ring,PACKET3(PACKET3_CP_DMA,4));
+		radeon_ring_write(ring,lower_32_bits(src_offset));
+		radeon_ring_write(ring,upper_32_bits(src_offset) & 0xFF);
+		radeon_ring_write(ring,lower_32_bits(dst_offset));
+		radeon_ring_write(ring,upper_32_bits(dst_offset) & 0xFF);
+		radeon_ring_write(ring,(cur_size_in_dw*4));*/
```

Comment:
Could these disabled ring writes be the cause of Jeff's slightly garbled textures from this comment?
https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/4#issuecomment-1101606320